src/components/PostDetails.tsx

```diff
@@ -1,117 +1,133 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { getComments, deleteComment } from '../api/posts';
+
+type Props = {
+  activePost: Post,
+};
+
+export const PostDetails: React.FC<Props> = ({ activePost }) => {
+  const [isFindComments, setIsFindComments] = useState(false);
+  const [postComments, setPostComments] = useState<Comment[] | []>([]);
+  const [isError, setIsError] = useState(false);
+  const [isFormActive, setIsFormActive] = useState(false);
+  const getPostComments = async (post: Post) => {
+    setIsError(false);
+    setIsFindComments(true);
+
+    try {
+      const response = await getComments(post.id);
+
+      setPostComments(response);
+
+      setIsFindComments(false);
+    } catch {
+      setIsFindComments(false);
+      setIsError(true);
+    }
+
+    setIsFindComments(false);
+  };
+
+  const removeComment = async (comment: Comment) => {
+    setPostComments(
+      postComments.filter(activeComment => activeComment.id !== comment.id),
+    );
+
+    deleteComment(comment.id, activePost.id);
+  };
+
+  useEffect(() => {
+    setPostComments([]);
+    setIsFormActive(false);
```

Comment:
```suggestion
```
default values

Reply:
`setIsFormActive` is still in `useEffect` callback
why do you need it there (you're setting the same value - redundant action)?